controllers/storagecluster/external_ocs_heartbeat.go

```diff
@@ -0,0 +1,70 @@
+package storagecluster
+
+import (
+	"context"
+	"os"
+
+	ocsv1 "github.com/red-hat-storage/ocs-operator/api/v1"
+	"github.com/red-hat-storage/ocs-operator/controllers/util"
+	externalClient "github.com/red-hat-storage/ocs-operator/services/provider/client"
+
+	batchv1 "k8s.io/api/batch/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+const StorageClusterNameEnvVar = "STORAGE_CLUSTER_NAME"
+
+func (r *StorageClusterReconciler) enableHeartBeatCronJob(instance *ocsv1.StorageCluster, externalClusterClient *externalClient.OCSProviderClient) error {
+	cronJob := &batchv1.CronJob{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "report-status-to-provider",
+			Namespace: instance.Namespace,
+		},
+	}
+
+	_, err := controllerutil.CreateOrUpdate(context.TODO(), r.Client, cronJob, func() error {
+		cronJob.Spec = batchv1.CronJobSpec{
+			Schedule: "* * * * *",
+			JobTemplate: batchv1.JobTemplateSpec{
+				Spec: batchv1.JobSpec{
+					Template: corev1.PodTemplateSpec{
+						Spec: corev1.PodSpec{
+							Containers: []corev1.Container{
+								{
+									Name:  "heartbeat",
+									Image: os.Getenv(providerAPIServerImage),
+									Command: []string{
+										"/usr/local/bin/status-reporter",
+									},
+									Env: []corev1.EnvVar{
+										{
+											Name:  util.WatchNamespaceEnvVar,
+											Value: os.Getenv(util.WatchNamespaceEnvVar),
+										},
+										{
+											Name:  "STORAGE_CLUSTER_NAME",
+											Value: instance.Name,
+										},
+									},
+								},
+							},
+							RestartPolicy:      corev1.RestartPolicyOnFailure,
+							ServiceAccountName: "ocs-status-reporter",
+						},
+					},
+				},
+			},
+		}
+		if err := controllerutil.SetOwnerReference(instance, cronJob, r.Client.Scheme()); err != nil {
+			return err
+		}
+
+		return nil
+	})
+	if err != nil {
+		return err
+	}
+	return nil
```

Comment:
just return err no need to have a check here?

Reply:
+1